pkg/skaffold/kubernetes/colorpicker_test.go

```diff
@@ -30,46 +31,43 @@ func TestColorPicker(t *testing.T) {
 		expectedColor color.Color
 	}{
 		{
-			description:   "not found",
-			pod:           &v1.Pod{},
-			expectedColor: color.None,
-		},
-		{
-			description: "found",
+			description: "not found",
 			pod: &v1.Pod{
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						{Image: "image"},
-					},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "unknown",
 				},
 			},
-			expectedColor: colorCodes[0],
+			expectedColor: color.None,
 		},
 		{
-			description: "ignore tag",
+			description: "found",
 			pod: &v1.Pod{
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						{Image: "image:tag"},
-					},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod",
 				},
 			},
 			expectedColor: colorCodes[0],
 		},
 		{
-			description: "second image",
+			description: "second pod",
 			pod: &v1.Pod{
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						{Image: "second:tag"},
-					},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "second",
 				},
 			},
 			expectedColor: colorCodes[1],
 		},
 	}
 
-	picker := NewColorPicker([]string{"image", "second"})
+	picker := NewColorPicker()
+	// register "second" twice and still expect colorCodes[1]
+	for _, v := range []string{"pod", "second", "second"} {
+		picker.Register(&v1.Pod{
```

Comment:
`Register` and `Pick` now involve locking. However, the current tests do not assert on concurrent use. Do we need that?